addons/resource/models/resource_calendar_attendance.py

```diff
@@ -64,6 +69,16 @@ def get_week_type(self, date):
         # some years have 53 weeks. Therefore, two consecutive odd week number follow each other (53 --> 1).
         return int(math.floor((date.toordinal() - 1) / 7) % 2)
 
+    @api.depends('hour_from', 'hour_to')
+    def _compute_duration_hours(self):
+        for attendance in self:
+            attendance.duration_hours = (attendance.hour_to - attendance.hour_from) if attendance.day_period != 'lunch' else 0
+
+    @api.depends('day_period')
+    def _compute_duration_days(self):
+        for attendance in self:
+            attendance.duration_days = 0.5 if attendance.day_period != 'lunch' else 0
```

Comment:
Shouldn't it be computed from the number of hours on the same day ?

Exemple:
Only 8 - 16 : 1 day
8 - 12 and 13 - 17 : 0.5 and 0.5
8 - 10 and 11 - 17 : 0.25 and 0.75

It seems a little bit simplistic currently, and you have to hardcode the values in the test etc.